src/main/java/com/cleanroommc/groovyscript/compat/mods/rustic/CrushingTub.java

```diff
@@ -0,0 +1,114 @@
+        addScripted(recipe);
+    }
+
+    public boolean remove(ICrushingTubRecipe recipe) {
+        addBackup(recipe);
+        return Recipes.crushingTubRecipes.remove(recipe);
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByOutput", example = {@Example("fluid('ironberryjuice')"), @Example("item('minecraft:sugar')")})
+    public boolean removeByOutput(IIngredient output) {
+        return Recipes.crushingTubRecipes.removeIf(entry -> {
+            if (output.test(entry.getResult()) || output.test(entry.getByproduct())) {
+                addBackup(entry);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput", example = @Example("item('rustic:wildberries')"))
+    public boolean removeByInput(IIngredient input) {
+        return Recipes.crushingTubRecipes.removeIf(entry -> {
+            if (input.test(entry.getInput())) {
+                addBackup(entry);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeAll", priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        Recipes.crushingTubRecipes.forEach(this::addBackup);
+        Recipes.crushingTubRecipes.clear();
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.streamRecipes", type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<ICrushingTubRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(Recipes.crushingTubRecipes).setRemover(this::remove);
+    }
+
+    @Property(property = "input", valid = @Comp("1"))
+    @Property(property = "fluidOutput", valid = @Comp("1"))
+    public static class RecipeBuilder extends AbstractRecipeBuilder<ICrushingTubRecipe> {
+
+        @Property
+        private ItemStack byproduct = ItemStack.EMPTY;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder byproduct(ItemStack byproduct) {
+            this.byproduct = byproduct;
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Rustic Crushing Tub recipe";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 1, 1, 0, 0);
+            validateFluids(msg, 0, 0, 1, 1);
+        }
+
+        @Override
+        @RecipeBuilderRegistrationMethod
+        public @Nullable ICrushingTubRecipe register() {
+            if (!validate()) return null;
+            ICrushingTubRecipe recipe = null;
+            for (ItemStack itemStack : input.get(0).getMatchingStacks()) {
+                ICrushingTubRecipe recipe1 = new CrushingTubRecipe(fluidOutput.get(0), itemStack, byproduct);
+                ModSupport.RUSTIC.get().crushingTub.add(recipe1);
+                if (recipe == null) recipe = recipe1;
+            }
+            return recipe;
```

Comment:
This is fine for now, but im thinking about returning a list of all added recipes. We have something like this in a few more places.